doc/mgmt/SONiC_OC_Vlan_HLD.md

```diff
@@ -0,0 +1,302 @@
+sonic(conf-if-Ethernet4)# no switchport access Vlan
+```
+
+#### 3.2.2.2 Show Commands
+#### Display VLAN Members detail
+`show Vlan`
+```
+Q: A - Access (Untagged), T - Tagged
+    NUM       Status       Q Ports
+    5         Active       T Ethernet24
+    10        Inactive
+    20        Inactive     A Po20
+```
+#### Display specific VLAN Members detail
+`show Vlan <vlan-id>`
+```
+sonic# show Vlan 5
+Q: A - Access (Untagged), T - Tagged
+    NUM    Status     Q Ports
+    5      Active     T Ethernet24
+                      T Po10
+                      A Ethernet20
+```
+#### Display VLAN information
+`show interface Vlan`
+```
+sonic# show interface Vlan
+Vlan10 is up, line protocol is up
+IP MTU 2500 bytes
+
+Vlan20 is up, line protocol is down
+IP MTU 5500 bytes
+```
+#### Display specific VLAN Information
+`show interface Vlan <vlan-id>`
+```sonic# show interface Vlan 10
+Vlan10 is up, line protocol is up
+IP MTU 2500 bytes
+```
+
+
+#### 3.2.2.3 Debug Commands
+N/A
+
+#### 3.2.2.4 IS-CLI Compliance
+N/A
+
+### 3.2.3 REST API Support
+
+**PATCH**
+- `/openconfig-interfaces:interfaces/ interface={name}`
+- `/openconfig-interfaces:interfaces/ interface={name}/config/[enabled | mtu]`
+- `/openconfig-interfaces:interfaces/interface={name}/openconfig-if-ethernet:ethernet/openconfig-vlan:switched-vlan/config/[access-vlan | trunk-vlans | interface-mode]`
+
+
+**DELETE**
+- `/openconfig-interfaces:interfaces/interface={name}/openconfig-if-ethernet:ethernet/openconfig-vlan:switched-vlan/config/[access-vlan | trunk-vlans]`
+- `/openconfig-interfaces:interfaces/ interface={name}`
+
+**GET**
+- `/openconfig-interfaces:interfaces/ interface={name}/state/[admin-status | mtu | oper-status]`
+- `/openconfig-interfaces:interfaces/interface={name}/openconfig-if-ethernet:ethernet/openconfig-vlan:switched-vlan/State/[access-vlan | trunk-vlans]`
+
+# 4 Flow Diagrams
+N/A
+
+# 5 Error Handling
+TBD
+
+# 6 Serviceability and Debug
+TBD
+
+# 7 Warm Boot Support
+N/A
+
+# 8 Scalability
+N/A
+
+# 9 Unit Test
+#### Configuration and Show via CLI
+
+| Test Name | Test Description |
+| :------ | :----- |
+| Create VLAN | Verify VLAN is configured |
+| Configure MTU for VLAN | Verify MTU is configured |
+| Remove MTU for VLAN | Verify MTU is reset to default |
+| Configure access VLAN  | Verify access VLAN is configured |
+| Configure trunk VLAN with access VLAN Id | Error should be thrown |
+| Configure access VLAN again | Verify access VLAN is present |
+| Shutdown VLAN | Verify VLAN is Inactive |
+| Enable VLAN | Verify VLAN is active |
+| Configure trunk VLAN to 2 different ports | Verify trunk VLAN is configured |
+| Configure access VLAN with trunk VLAN Id | Error should be thrown |
+| Configure trunk VLAN again | Verify trunk VLAN is present |
+| Remove access VLAN | Verify access VLAN is removed |
+| Remove trunk VLAN | Verify trunk VLANs are removed |
+| Delete an Invalid VLAN | Error should be thrown |
+| Delete the VLAN | Verify the VLAN is deleted |
```

Comment:
We should reject and throw error if there is any Vlan member present for that vlan.

Reply:
If we reject Vlan deletion with members in it, then user needs to go to each member & remove the config - can be a task with multiple ports in Vlan. Hence we are leaning towards implicit deletion of member association when Vlan is deleted.